src/components/JournalRerouter/index.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import settings from '../../data/configuration/constants';
+
+import { Route, Redirect } from 'react-router';
+
+class JournalRerouter extends React.Component {
+  componentDidMount(){
+    this.props.getJournal(this.props.match.params.journalId);
+  }
+  render(){
+    if(this.props.userInfoFinishedFetching && this.props.journalFinishedFetching){
+      // Only run if both the user info and journal info API calls have finished
+      if(this.props.loggedIn) {
+        if(this.props.lastVisitedPage){
+          // Send user to the last page they visited if available
+          let lastVisitedPageUrl = `/${this.props.match.params.journalId}/pages/${this.props.lastVisitedPage}`;
```

Comment:
where do we set the lastVisitedPage?

Reply:
https://github.com/edx/journals-frontend/pull/3/files#diff-d9c51a596f54d7ad40565b79574689f8R15
`lastVisitedPage: state.userInfo.visitedPages.length != 0 ? state.userInfo.visitedPages[0].page: null`

It's just taking the PageVisit items returned and grabbing the first one.

Reply:
@tuchfarber 
right but don't think we are setting the PageVisits for a given user anywhere. This needs to be an explicit post to the pagevisit endpoint by the client the way it's currently implemented. See https://github.com/edx/journals/blob/master/journals/apps/api/v1/tests/test_views.py#L28

Reply:
ack